client/components/category-nav/index.tsx

```diff
@@ -0,0 +1,105 @@
+import { Button } from '@wordpress/components';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import classnames from 'classnames';
+
+import './style.scss';
+
+type CategoryNavProps = {
+	buttons?: {
+		icon: string;
+		label: string;
+		link: string;
+	}[];
+	children?: React.ReactNode;
+	list: {
+		name: string;
+		label?: string;
+		link: string;
+	}[];
+	selectedCategory: string;
+};
+
+export const CategoryNav = ( { buttons, children, list, selectedCategory }: CategoryNavProps ) => {
```

Comment:
We usually avoid abbreviations (see for example `NavigationHeader`), so I would recommend renaming that component to `CategoryNavigation`. Since we're introducing a new component that can be used in different places, what do you think of adding it to the [Calypso Docs](https://wpcalypso.wordpress.com/devdocs) to make it more discoverable? Finally, we could probably add a readme file that describes roughly what that component does (and maybe with a screenshot).

Reply:
>Since we're introducing a new component that can be used in different places, what do you think of adding it to the https://wpcalypso.wordpress.com/devdocs to make it more discoverable?

Yep, it's my plan for the follow up PR. Since we still don't have final designs, and additionally I have no idea how to add components to design sysyem, and also necessary to spend some time on description, screenshots etc. 

> We usually avoid abbreviations

ON the one hand it's better to keep consistency, but on the other hand I agree - we have to avoid old mistakes with `Nav`, so I renamed it, thanks.

Reply:
Given that we plan to reuse this nav, it does make sense to implement it as a generalized component. I would consider tweaking the name to something like `CategoryPillNavigation`, which I think is more descriptive.

Reply:
I don't mind such a name, renamed